### ISSUE TITLE
bcbio pipeline stopping without error message

### ISSUE THREAD

**Issue by Wixaros** [chapmanb/bcbio-nextgen#1767]
Hi, 

We are trying to get the bcbio install to work on our clusters, however we are experiencing some issues.

Our clusters are running with GRID engine.

I have attached the three logs and our yaml file to show what we are currently doing. It seems that the program halts at the "goleft depth" command. It simply stops and doesn't continue.

Our shell script for starting bcbio looks like this and is executed from the "work" directory within the project directory as stated in the manual.
```
#$ -S /bin/sh
#$ -l h_vmem=4.0G
#$ -cwd
bcbio_nextgen.py ../config/OCCC_project_S1.yaml
```

Log files:
[bcbio-nextgen.log.txt](https://github.com/chapmanb/bcbio-nextgen/files/716972/bcbio-nextgen.log.txt)
[bcbio-nextgen-commands.log.txt](https://github.com/chapmanb/bcbio-nextgen/files/716973/bcbio-nextgen-commands.log.txt)
[bcbio-nextgen-debug.log.txt](https://github.com/chapmanb/bcbio-nextgen/files/716974/bcbio-nextgen-debug.log.txt)

Yaml config file for one sample in the project
[OCCC_project_S1.yaml.txt](https://github.com/chapmanb/bcbio-nextgen/files/716978/OCCC_project_S1.yaml.txt)

We have been struggling quite some time wit this, and are still having some issues, however we are very excited about BCbio and hope to get it working.

**Comment by chapmanb** [comment]
Thanks for trying out bcbio and apologies about the issue. You're right that there are no informative messages in these logs to help debug what is going wrong. What behavior are you seeing from the submitted job during the run? Is it killed or otherwise stopped by grid engine? Are there any useful messages in the SGE log file that might give us more of an idea? It's possible that it triggers some kind of resource issue resulting in the scheduler killing it which might be reflected there. Hope this provides some clue for debugging further.

**Comment by Wixaros** [comment]
Hi again

We tried to post the shell script below  with the `qsub -r y` command

```
#$ -S /bin/sh
#$ -cwd
bcbio_nextgen.py ../config/OCCC_project_S1.yaml
```

and we recieved the following error message:
`/opt/gridengine/default/spool/compute-21-3/job_scripts/4867: line 3: 24604 Segmentation fault      (core dumped) bcbio_nextgen.py ../config/OCCC_project_S4.yaml
`

**Comment by chapmanb** [comment]
Thanks for helping to debug further. It seems from this command like `bcbio_nextgen.py` itself is having a segmentation fault, resulting in the sudden stop. I'm not sure why python/bcbio would segfault here. What type of system are you running this on? Are you able to use gdb to try and debug where the issue happens? If so, you'd need to figure out the location of the python used to run bcbio (which should be at the top of the `bcbio_nextgen.py` script. Then do:
```
$gdb /path/to/anaconda/bin/python
(gdb) run /path/to/bcbio_nextgen.py ../config/OCCC_project_S4.yaml
```
and when it segfaults get more information about what happened by looking at the traceback:
```
(gdb) bt
```
Happy to try and help more with additional details. Thanks for the help debugging.

**Comment by Wixaros** [comment]
Hi again

We have been working hard during the weekend, and we are happy to report that we have resolved the problems. It seemed that the configuration file (bcbio_system.yaml) wasn't correctly read so that our system didn't allocate the required memory, hence a memory problem. We fixed it with adding a manual stating that the memory to be used needs to be stated in the `qsub` command while executing it.

So basically we have set the yaml file configurations to the highest possible value on our cluster, and then we are able to adjust the settings with the qsub command. Works like a charm
